core/src/main/java/fr/sncf/osrd/infra/implementation/reservation/ReservationInfraBuilder.java

```diff
@@ -148,11 +149,11 @@ private ImmutableNetwork<DiDetector, ReservationRoute> makeRouteGraph(
 
     /** Checks that the route makes sense, reports any relevant warning or error otherwise */
     private void validateRoute(ReservationRouteImpl route, RJSRoute rjsRoute) {
-        if (route.getDetectorPath().isEmpty()) {
+        if (route.getDetectorPath().size() < 2) {
             warningRecorder.register(new Warning(String.format(
-                    "Route %s has no detector on its track ranges", rjsRoute.id
+                    "Route %s has less than two detectors on its track ranges", rjsRoute.id
```

Comment:
What about getting this message from MissingDetectorsRoute ?

Reply:
I changed the behavior of the recorder so we can trigger multiple errors then verify and throw all of them at the same time